packages/lit-dev-cloudbuild-cleanup/src/cleanup.ts

```diff
@@ -0,0 +1,313 @@
+      openPrs.add(pr.number);
+    }
+  }
+
+  console.log('Found open PRs', openPrs);
+
+  const run = google.run('v1');
+
+  const auth = new google.auth.GoogleAuth({
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  });
+
+  const authClient = await auth.getClient();
+  google.options({auth: authClient});
+
+  // Fetch list of revisions across both services.
+  // This API provides the revision's created time but does not contain data
+  // of any associated traffic tags.
+  console.log('Fetching list of revisions');
+  const {data} = await run.projects.locations.revisions.list({
+    parent: `projects/${PROJECT_ID}/locations/${_DEPLOY_REGION}`,
+  });
+
+  let revisions = data;
+
+  if (!revisions.items) {
+    // We should never reach this since there should always be running revisions
+    // for the deployed services.
+    throw new Error('Found no revision items');
+  }
+
+  console.log('Total revisions count', revisions.items.length);
+
+  const revisionsToKeep = new Set<string>();
+  for (const rev of revisions.items) {
+    if (
+      !rev.metadata ||
+      !rev.metadata.creationTimestamp ||
+      !rev.metadata.name
+    ) {
+      throw new Error(
+        `Found revision with missing necessary metadata: ${JSON.stringify(rev)}`
+      );
+    }
+
+    if (new Date(rev.metadata.creationTimestamp) > ONE_WEEK_AGO) {
+      revisionsToKeep.add(rev.metadata.name);
+    }
+  }
+
+  console.log('Revisions younger than a week count', revisionsToKeep.size);
+
+  // Fetch each service and identify additional revisions to keep based on traffic tag.
+  // Also update service with filtered traffic array if needed.
+  const services = ['lit-dev', 'lit-dev-playground'];
+  let serviceUpdated = false;
+  for (const serviceName of services) {
+    console.log(`Fetching ${serviceName} service`);
+    const {data} = await run.projects.locations.services.get({
+      name: `projects/${PROJECT_ID}/locations/${_DEPLOY_REGION}/services/${serviceName}`,
+    });
+
+    if (!data.spec || !data.spec.traffic) {
+      throw new Error(`Found no traffic for ${serviceName} service`);
+    }
+
+    console.log(`${serviceName} traffic count`, data.spec.traffic.length);
+
+    const trafficToKeep = data.spec.traffic.filter((t) => {
+      if (!t.revisionName) {
+        throw new Error('Found traffic with no revision name');
+      }
+
+      // The set already has revisions to keep based on recency
+      if (revisionsToKeep.has(t.revisionName)) {
+        return true;
+      }
+
+      // Presence of percent > 0 indicates traffic is being routed here
+      // i.e. revision is used for current prod deployment
+      if (t.percent) {
+        revisionsToKeep.add(t.revisionName);
+        return true;
+      }
+
+      if (t.tag) {
+        // Tags should look like "pr529-8b0837f" or "main-45ace0e"
+        if (t.tag.startsWith('pr')) {
+          // pr529-8b0837f
+          //   ^^^
+          const prNumber = parseInt(t.tag.slice(2), 10);
```

Comment:
Seems like this is depending on the behavior that `parseInt` ignores extra noise, (e.g. `parseInt("529-8b0837f") // 529`).

I think a regexp might be more explicit/clear:

```ts
const prMatch = t.tag.match(/^pr(\d+)-/);
if (prMatch) {
  const prNumber = Number(t.tag[1]);
  ...
} else if (!t.tag.startsWith('main')) {
  ...
}
```